File: StructuralPatterns/Facade/Electronics.py
class Electronics:
    def __init__(self):
        self.power = False

    def power_on(self):
        self.power = True

    def power_off(self):
        self.power = False

    def is_power_on(self):
        return self.power

    def show_status(self):
        if self.power:
            print(type(self).__name__ + " is working ")
        else:
            print(type(self).__name__ + " not work ")


class KTVSystem(Electronics):
    def __init__(self):
        super().__init__()
        self.song = ""

    def select_song(self, song):
        self.song = song

    def play_song(self):
        print(" play song: " + self.song)


class PS3(Electronics):
    def __init__(self):
        super().__init__()
        self.CD = ""

    def put_CD(self, CD):
        self.CD = CD

    def play(self):
        print(" play CD: " + self.CD)

    def show_status(self):
        super().show_status()
        if self.is_power_on():
            print(" play CD: " + self.CD)

File: DESIGN-CSharp/Facade/Electronics.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Facade
{
    public abstract class Electronics
    {
        private bool power = false;

        public void powerOn()
        {
            this.power = true;
        }

        public void powerOff() 
        {
            this.power = false;
        }

        public bool isPowerOn() 
        {
            return this.power;
        }

        public virtual void showStatus() 
        {
            if (this.power)
            {
                Console.WriteLine(this.GetType().Name + " is working ");
            }
            else 
            {
                Console.WriteLine(this.GetType().Name + " not work ");
            }
        }
    }

    public class KTVSystem : Electronics 
    {
        private string song { get; set; }

        public void selectSong(string song) 
        {
            this.song = song;
        }

        public void playSong() 
        {
            Console.WriteLine(" play song: " + this.song);
        }
    }

    public class PS3 : Electronics 
    {
        private string CD { get; set; }

        public void putCD(string CD) 
        {
            this.CD = CD;
        }

        public void play() 
        {
            Console.WriteLine(" play CD: " + this.CD);
        }

        public override void showStatus() 
        {
            base.showStatus();
            if (isPowerOn()) 
            {
                Console.WriteLine(" play CD: " + this.CD);
            }
        }
    }
}
